cadc-vos-server/src/main/java/org/opencadc/vospace/server/pkg/VospacePackageRunner.java

```diff
@@ -0,0 +1,379 @@
+import org.opencadc.vospace.ContainerNode;
+import org.opencadc.vospace.DataNode;
+import org.opencadc.vospace.LinkNode;
+import org.opencadc.vospace.LinkingException;
+import org.opencadc.vospace.Node;
+import org.opencadc.vospace.NodeNotFoundException;
+import org.opencadc.vospace.VOS;
+import org.opencadc.vospace.VOSURI;
+import org.opencadc.vospace.server.NodeFault;
+import org.opencadc.vospace.server.NodePersistence;
+import org.opencadc.vospace.server.PathResolver;
+import org.opencadc.vospace.server.Utils;
+import org.opencadc.vospace.server.auth.VOSpaceAuthorizer;
+import org.opencadc.vospace.server.transfers.TransferGenerator;
+import org.opencadc.vospace.transfer.Protocol;
+import org.opencadc.vospace.transfer.Transfer;
+import org.opencadc.vospace.transfer.TransferParsingException;
+import org.opencadc.vospace.transfer.TransferReader;
+
+public class VospacePackageRunner extends PackageRunner {
+
+    private static final Logger log = Logger.getLogger(VospacePackageRunner.class);
+    private static final String CONTINUING_PROCESSING = ", continuing package processing...";
+    private URI resourceID;
+    private Transfer packageTransfer;
+    private NodePersistence nodePersistence;
+    private VOSpaceAuthorizer vospaceAuthorizer;
+    private PathResolver pathResolver;
+
+    public VospacePackageRunner() {}
+
+    /**
+     * Sets the name of this instance which is used to generate the JNDI key to the NodePersistence,
+     * and initialize other resources used by this class.
+     *
+     * @param appName the name of the instance.
+     */
+    @Override
+    public void setAppName(String appName) {
+        String jndiKey = appName + "-" + NodePersistence.class.getName();
+        try {
+            Context ctx = new InitialContext();
+            this.nodePersistence = (NodePersistence) ctx.lookup(jndiKey);
+            this.vospaceAuthorizer = new VOSpaceAuthorizer(nodePersistence);
+            this.pathResolver = new PathResolver(nodePersistence, vospaceAuthorizer, true);
+            this.resourceID = nodePersistence.getResourceID();
+        } catch (NamingException e) {
+            throw new RuntimeException("BUG: NodePersistence implementation not found with JNDI key " + jndiKey, e);
+        }
+    }
+
+    /**
+     * Get the list of targets for the package and the package name.
+     */
+    @Override
+    protected void initPackage() {
+        log.debug("initPackage start");
+        try {
+            // check job is valid
+            log.debug("job id passed in: " + job.getID());
+            JobInfo jobInfo = job.getJobInfo();
+
+            // Get the target list from the job
+            TransferReader tr = new TransferReader();
+            this.packageTransfer = tr.read(jobInfo.getContent(), VOSURI.SCHEME);
+            List<URI> targetList = packageTransfer.getTargets();
+            if (targetList.size() > 1) {
+                this.packageName = "cadc-download-" + job.getID();
```

Comment:
instead of `cadc-download-` this should use appname (eg cavern or arc or vault), eg `appname + "-download-"` as the base name... then append jobID or name as approp.

Reply:
Updated the package name.